pkg/kubectl/cmd/util/factory.go

```diff
@@ -354,6 +354,19 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 					return nil, errors.New("provided options object is not a PodLogOptions")
 				}
 				return c.Pods(t.Namespace).GetLogs(t.Name, opts), nil
+
+			case *api.ReplicationController:
+				opts, ok := options.(*api.PodLogOptions)
+				if !ok {
+					return nil, errors.New("provided options object is not a PodLogOptions")
+				}
+				selector := labels.SelectorFromSet(t.Spec.Selector)
+				pod, err := GetFirstPod(c, t.Namespace, selector)
```

Comment:
It looks like `attach` only acts on rcs as an option of `kubectl run`. The `--interactive` flag throws an error if the number of replicas is not 1, but I think it should be doing the same for `--attach`. The help text indicates it expects a single pod, not the first of many. Might be a missed case?

cc @bgrant0607